File: src/DynamicLinkLibraries/3DConverters/Abstract3DConverters/EmissiveMaterial.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Abstract3DConverters
{
    public class EmissiveMaterial : SimpleMaterial
    {
        public Image Image { get; private set; }

        public EmissiveMaterial(Color color, Image image = null) : base(color)
        {
            if (image != null)
            {
                Image = image.Clone() as Image;
            }
        }

        protected override object CloneIfself()
        {
            return new EmissiveMaterial(Color.Clone() as Color, null);
        }
    }
}

File: src/Web/reactprojectsimplegame/src/Library/Abstract3DConverters/Materials/EmissiveMaterial.ts

import { ColorTexture } from "../ColorTexture";
import { ImageTexture } from "../ImageTexture";
import { SimpleMaterial } from "./SimpleMaterial";

export class EmissiveMaterial extends SimpleMaterial {
    constructor(name: string, color: ColorTexture, image: ImageTexture) {
        super(name, color)
        this.image = image
    }

}